compiler/semobjconstr.nim

```diff
@@ -398,8 +398,12 @@ proc semObjConstr(c: PContext, n: PNode, flags: TExprFlags; expectedType: PType
       # multiple times as long as they don't have closures.
       result.typ.flags.incl tfHasOwned
   if t.kind != tyObject:
-    return localErrorNode(c, result,
-      "object constructor needs an object type".dup(addDeclaredLoc(c.config, t)))
+    return localErrorNode(c, result, if t.kind != tyGenericBody:
+      "object constructor needs an object type".dup(addDeclaredLoc(c.config, t))
+      else: "cannot instantiate: '" &
+        typeToString(t, preferDesc) &
+        "'; Maybe generic arguments are missing?"
```

Comment:
```suggestion
        "'; Are generic arguments missing?"
```

Reply:
it's copyed from compiler\semtypinst.nim replaceTypeVarsTAux, am not native english speaker, dunno which is better, make choice then change these two place ?

Reply:
Can you give me a more thorough explanation of when this error occurs? That would give me a better idea of how this error should be written.

Reply:
it happened 
1, use object constructor directly without generic type
2, user custom constructor and return type lose generic type

Reply:
Ok, then I think the best phrasing here would be
```suggestion
        "'; the object's generic parameters cannot be inferred and must be explicitly given"
```